gitlab/utils.py

```diff
@@ -71,18 +71,22 @@ def _transform_types(
     data = data.copy()
     files = {}
 
-    for attr_name, type_cls in custom_types.items():
+    for attr_name, attribute_class in custom_types.items():
         if attr_name not in data:
             continue
 
-        type_obj = type_cls(data[attr_name])
-
-        # if the type if FileAttribute we need to pass the data as file
-        if transform_files and isinstance(type_obj, types.FileAttribute):
-            key = type_obj.get_file_name(attr_name)
-            files[attr_name] = (key, data.pop(attr_name))
-        else:
-            data[attr_name] = type_obj.get_for_api()
+        gitlab_attribute = attribute_class(data[attr_name])
+        if isinstance(gitlab_attribute, types.GitlabAttribute):
+            # if the type is FileAttribute we need to pass the data as file
+            if isinstance(gitlab_attribute, types.FileAttribute):
+                if transform_files:
+                    key = gitlab_attribute.get_file_name(attr_name)
+                    files[attr_name] = (key, data.pop(attr_name))
+                continue
+            key, value = gitlab_attribute.get_for_api(key=attr_name)
```

Comment:
Haven't explored at all why, but pylance isn't happy here. :shrug: I guess it doesn't matter.
![image](https://user-images.githubusercontent.com/16777978/180972772-b0658599-95cb-4474-8170-9ba613c43f6b.png)